Frontend/RecRock-Calendar-Frontend/src/app/calendar/components/calendar/calendar.component.ts

```diff
@@ -49,50 +50,28 @@ export class CalendarComponent {
   }
 
   private openDialog(date: Date): void {
-    let userEvent: Event | undefined = this.events.find((event) =>
+    const userEvent: Event | undefined = this.events.find((event) =>
       event.isEventEqual(this.User.name, date)
     );
 
-    //---------------------------------------------------------------------------------------
-    //TODO: Refactor this part
-    let startTime: Date;
-    let endTime: Date;
-    let eventExists: boolean;
-    if (userEvent !== undefined) {
-      this.events = this.events.filter((event) => event !== userEvent);
-      startTime = userEvent.start;
-      endTime = userEvent.end;
-      eventExists = true;
-    } else {
-      startTime = new Date();
-      endTime = new Date();
-      eventExists = false;
-    }
-    //---------------------------------------------------------------------------------------
-
-    let popupObject = new PopupDataObject(this.User.name, this.formatTime(startTime), this.formatTime(endTime), date, eventExists, false)
-
     const dialogRef = this.dialog.open(EditDayPopupComponent, {
-      data: popupObject
+      data: this.initialisePopupDTO(userEvent, date),
     });
 
     dialogRef.afterClosed().subscribe((result) => {
-      //---------------------------------------------------------------------------------------
-      //TODO: Refactor this part
-      if (result) {
-        if (result.deleteEvent && userEvent !== undefined) {
+      if (userEvent !== undefined) {
+        if (result.deleteEvent) {
           this.deleteEvent(userEvent);
-        } else {
-          let startDate: Date = this.addTimeToDate(result.startTime, date);
-          let endDate: Date = this.addTimeToDate(result.endTime, date);
-          this.addEvent(startDate, endDate);
         }
-      } else {
-        if (userEvent !== undefined) {
+        if (result.noModification) {
           this.events = [...this.events, userEvent];
         }
       }
-      //---------------------------------------------------------------------------------------
+      if (result.saveEvent) {
+        let startDate: Date = this.addTimeToDate(result.startTime, date);
+        let endDate: Date = this.addTimeToDate(result.endTime, date);
+        this.addEvent(startDate, endDate);
+      }
     });
   }
 
```

Comment:
I would create a folder called "helpers" in the shared module and I would add a date-time.helper.ts file. If you add the addTimeToDate method there, it will be much easier in the future to reuse it.